azurerm/internal/services/storage/resource_arm_storage_blob.go

```diff
@@ -115,6 +115,13 @@ func resourceArmStorageBlob() *schema.Resource {
 				ConflictsWith: []string{"source", "source_content"},
 			},
 
+			"content_md5": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"source_uri"},
```

Comment:
Is there any ValidateFunc which ensure that `content_md5` is not set when blob type is other than block? If yes, I can get rid of some validations from the code.

Reply:
Hi @Humoiz 

Just a quick note while I'm passing through, my understanding is that the Azure Blob MD5 and the Terraform MD5 values are not necessarily the same.  Azure has a slightly more involved route to the value, as covered [in depth here](https://docs.microsoft.com/en-us/archive/blogs/windowsazurestorage/windows-azure-blob-md5-overview)

I believe an alternative checksum mechanism is coming (though I can't find a link to where I read that)

in response to your points while I'm here:

1. This would need to be handled in the CRUD code, not the schema. ValidateFunc is explicitly to check the value being supplied for the property, not the context it's used in, this would need to be checked when preparing the data for send.
2. See above, it should be possible to duplicate the Azure path to the B64 encoded sum, but I've not looked in any depth at it. There is no current implementation I am aware of for Terraform.

Reply:
Thanks for the feedback. I understand the terraform MD5 and Azure Blob MD5 is different. I have followed the conversation [here](https://github.com/hashicorp/terraform/issues/19245) to implement this solution. I`m expecting users to provide Terraform MD5 and under the hood I am converting the Hex to base64 encoded. 

1. Currently this is handled in the code, I was just curious to know if this can be handled via ValidateFunc.